### ISSUE TITLE
Exception with android lint enabled - Cannot access first() element from an empty List

### ISSUE THREAD

**Issue by kdani41** [btkelly/gnag#173]
**Version** - 2.0.0
**Gradle lint options**
```gradle
  lintOptions {
    lintConfig file("$project.rootDir/tools/rules-lint.xml")
    htmlOutput file("$project.buildDir/outputs/lint/lint.html")
    warningsAsErrors true
    xmlReport false
    abortOnError false
  }

  gnag {
     androidLint {
       enabled true
       severity 'Error'
    }
  }
```

Error log on `./gradlew clean gnagCheck --stacktrace`
```java 
> Task :app:gnagCheck
Checkstyle detected 400 violations.
PMD detected 20 violations.


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:gnagCheck'.
> Cannot access first() element from an empty List

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:gnagCheck'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:100)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:63)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:88)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:124)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:80)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:105)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:99)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:625)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:580)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:99)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: java.util.NoSuchElementException: Cannot access first() element from an empty List
        at com.btkelly.gnag.reporters.AndroidLintViolationDetector.reportFile(AndroidLintViolationDetector.groovy:107)
        at com.btkelly.gnag.reporters.AndroidLintViolationDetector.isEnabled(AndroidLintViolationDetector.groovy:47)
        at com.btkelly.gnag.tasks.GnagCheck$$Lambda$27925/894768514.test(Unknown Source)
        at com.btkelly.gnag.tasks.GnagCheck.executeGnagCheck(GnagCheck.java:95)
        at com.btkelly.gnag.tasks.GnagCheck.taskAction(GnagCheck.java:85)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.doExecute(DefaultTaskClassInfoStore.java:141)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:134)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:121)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:731)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:705)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:122)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:111)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
        ... 27 more

```

**Comment by stkent** [comment]
The failure occurs when Gnag tries to find the lint report file:  https://github.com/btkelly/gnag/blob/112f954/plugin/src/main/groovy/com/btkelly/gnag/reporters/AndroidLintViolationDetector.groovy#L107-L109

If you disable Gnag and run lint, where is your lint report file located? That information should help us diagnose the problem :)

**Comment by kdani41** [comment]
Here are my lint options

```gradle
  lintOptions {
    lintConfig file("$project.rootDir/tools/rules-lint.xml")
    htmlOutput file("$project.buildDir/outputs/lint/lint.html")
    warningsAsErrors true
    xmlReport false
    abortOnError false
  }
```

**Comment by stkent** [comment]
Ah; Gnag relies on the XML-format lint report file to parse lint errors, so if you enable that (by setting `xmlReport true`) you should be good to go. Try that out and let us know if it works. If it doesn't, we'll continue to triage; if it does, we can see if it's easy to produce a better error in this case :)

**Comment by kdani41** [comment]
Yes after changing `xmlReport true` it works :)

**Comment by stkent** [comment]
Brilliant; I'll close this issue out when we decide what steps to take to help avoid this in the future!

**Comment by stkent** [comment]
Closed by #179 :)

**Comment by stkent** [comment]
Will be in the next release!

**Comment by stkent** [comment]
Reopened until next release ships.

**Comment by stkent** [comment]
This was released with either v2.0.0 or v2.1.0.
